src/actions/account.js

```diff
@@ -18,9 +26,13 @@ export const accountUpdated = data => ({
  *
  * @returns {Object} - Action object
  */
-export const accountLoggedOut = () => ({
-  type: actionTypes.accountLoggedOut,
-});
+export const accountLoggedOut = () =>
+  (dispatch) => {
+    dispatch(forgingReset());
+    dispatch(activePeerReset());
+    dispatch(transactionsReset());
+    dispatch({ type: actionTypes.accountLoggedOut });
+  };
```

Comment:
Each dispatch will handle view re-render. I recommend to using one action that will be handled in different reduces. like, handle `logout` in each of it.